pkg/scalers/azure_eventhub_scaler.go

```diff
@@ -0,0 +1,230 @@
+package scalers
+
+import (
+	"context"
+	"fmt"
+	"math"
+	"strconv"
+
+	eventhub "github.com/Azure/azure-event-hubs-go"
+	"github.com/Azure/azure-storage-blob-go/azblob"
+	log "github.com/Sirupsen/logrus"
+	"k8s.io/api/autoscaling/v2beta1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/metrics/pkg/apis/external_metrics"
+)
+
+const (
+	defaultEventHubMessageThreshold = 10
+	eventHubMetricType              = "External"
+	thresholdMetricName             = "unprocessedEventThreshold"
+	defaultEventHubConsumerGroup    = "$Default"
+	defaultEventHubConnectionString = "EventHub"
+	defaultStorageConnectionString  = "AzureWebJobsStorage"
+)
+
+type AzureEventHubScaler struct {
+	metadata           *EventHubMetadata
+	client             *eventhub.Hub
+	storageCredentials *azblob.SharedKeyCredential
+}
+
+type EventHubMetadata struct {
+	eventHubConnection    string
+	eventHubConsumerGroup string
+	threshold             int64
+	storageConnection     string
+}
+
+// NewAzureEventHubScaler creates a new scaler for eventHub
+func NewAzureEventHubScaler(resolvedEnv, metadata map[string]string) (Scaler, error) {
+	parsedMetadata, err := parseAzureEventHubMetadata(metadata, resolvedEnv)
+	if err != nil {
+		return nil, fmt.Errorf("unable to get eventhub metadata: %s", err)
+	}
+
+	_, cred, err := GetStorageCredentials(parsedMetadata.storageConnection)
+	if err != nil {
+		return nil, fmt.Errorf("unable to get storage credentials: %s", err)
+	}
+
+	hub, err := GetEventHubClient(parsedMetadata.eventHubConnection)
+	if err != nil {
+		return nil, fmt.Errorf("unable to get eventhub client: %s", err)
+	}
+
+	return &AzureEventHubScaler{
+		metadata:           parsedMetadata,
+		storageCredentials: cred,
+		client:             hub,
+	}, nil
+}
+
+// parseAzureEventHubMetadata parses metadata
+func parseAzureEventHubMetadata(metadata, resolvedEnv map[string]string) (*EventHubMetadata, error) {
+	meta := EventHubMetadata{}
+	meta.threshold = defaultEventHubMessageThreshold
+
+	if val, ok := metadata[thresholdMetricName]; ok {
+		threshold, err := strconv.ParseInt(val, 10, 64)
+		if err != nil {
+			return nil, fmt.Errorf("Error parsing azure eventhub metadata %s: %s", thresholdMetricName, err)
+		}
+
+		meta.threshold = threshold
+	}
+
+	storageConnectionSetting := defaultStorageConnectionString
+	if val, ok := metadata["storageConnection"]; ok && val != "" {
+		storageConnectionSetting = val
+	} else {
+		return nil, fmt.Errorf("no storage connection setting given")
+	}
+
+	if val, ok := resolvedEnv[storageConnectionSetting]; ok {
+		meta.storageConnection = val
+	} else {
+		return nil, fmt.Errorf("no storage connection setting given")
```

Comment:
Can you change this from being the same error as the metadata["storageConnection"] ?